doc/_extensions/zephyr/api_overview.py

```diff
@@ -0,0 +1,179 @@
+# Copyright (c) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import importlib
+import importlib.util
+
+from docutils import nodes
+from pathlib import Path
+from sphinx.application import Sphinx
+from sphinx.util.docutils import SphinxDirective
+from sphinx.util import logging
+from typing import Any, Dict
+
+logger = logging.getLogger(__name__)
+
+doxmlparser = None
+if importlib.util.find_spec("doxmlparser"):
+    doxmlparser = importlib.import_module("doxmlparser")
+
+    from doxmlparser.compound import DoxCompoundKind
+else:
+    logger.warning(
+        "Module `doxmlparser` not found. API overview table will not be generated"
+    )
```

Comment:
If this package is not pip-installable, then you should parse Doxygen files yourself (it's not that complicated). Depending on something like this is a bad idea (no versioning, how to install, etc).

Reply:
we should just include this in our tree in scripts or docs/ or whatever.

Reply:
doesn't sound like a good idea by looking at https://github.com/doxygen/doxygen/tree/master/addon/doxmlparser. Maybe they should just publish this package. I remember this parser being very slow, to the point I discarded using it for https://github.com/zephyrproject-rtos/zephyr/pull/39702 has this been checked?

Reply:
for whatever we do here, that should be fast enough I think. if this becomes a package, we can move to it, but "slowness" will not go away when in pypi

Reply:
```python
    import time
    s = time.perf_counter()
    groups = parse_xml_dir(doxygen_xml_dir)
    logger.info(f"XML parsing took {time.perf_counter() - s:.2f} seconds")
```

> XML parsing took 10.67 seconds